homeassistant/components/myuplink/select.py

```diff
@@ -0,0 +1,98 @@
+"""Select entity for myUplink."""
+
+from aiohttp import ClientError
+from myuplink import DevicePoint
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import MyUplinkConfigEntry, MyUplinkDataCoordinator
+from .entity import MyUplinkEntity
+from .helpers import find_matching_platform, skip_entity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MyUplinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up myUplink select."""
+    entities: list[SelectEntity] = []
+    coordinator: MyUplinkDataCoordinator = config_entry.runtime_data
+
+    # Setup device point select entities
+    for device_id, point_data in coordinator.data.points.items():
+        for point_id, device_point in point_data.items():
+            if skip_entity(device_point.category, device_point):
+                continue
+            description = None
+            if find_matching_platform(device_point, description) == Platform.SELECT:
+                entities.append(
+                    MyUplinkSelect(
+                        coordinator=coordinator,
+                        device_id=device_id,
+                        device_point=device_point,
+                        entity_description=description,
+                        unique_id_suffix=point_id,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class MyUplinkSelect(MyUplinkEntity, SelectEntity):
+    """Representation of a myUplink select entity."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: SelectEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the select."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        # Internal properties
+        self.point_id = device_point.parameter_id
+        self._attr_name = device_point.parameter_name
+
+        self._attr_options = [x["text"].capitalize() for x in device_point.enum_values]
+        self.options_map = {
+            str(int(x["value"])): x["text"].capitalize()
+            for x in device_point.enum_values
+        }
+        self.options_rev = {
+            x["text"].capitalize(): str(int(x["value"]))
+            for x in device_point.enum_values
```

Comment:
Use the already calculated options dict to reverse the doct.

Reply:
Like this?